qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -54,14 +54,14 @@ class Command(misc.MinimalLineEditMixin, misc.CommandLineEdit):
     show_cmd = pyqtSignal()
     hide_cmd = pyqtSignal()
 
-    def __init__(self, win_id, parent=None):
-        misc.CommandLineEdit.__init__(self, parent)
+    def __init__(self, *, win_id, private, parent=None):
+        misc.CommandLineEdit.__init__(self, parent=parent)
         misc.MinimalLineEditMixin.__init__(self)
         self._win_id = win_id
-        command_history = objreg.get('command-history')
-        self.history.handle_private_mode = True
-        self.history.history = command_history.data
-        self.history.changed.connect(command_history.changed)
+        if not private:
+            command_history = objreg.get('command-history')
+            self.history.history = command_history.data
+            self.history.changed.connect(command_history.changed)
```

Comment:
I haven't tried this yet, so I have no idea how it looks like in practice, but I'm wondering if it would be useful to have a read-only access to the global history for the purpose of the history completion.